polynomial/__init__.py

```diff
@@ -180,31 +198,32 @@ def __ne__(self, other):
 
     def __bool__(self):
         """Return not self == 0."""
-        return not self == 0
+        return self != 0
 
     def __add__(self, other):
         """Return self + other."""
         if not self:
-            return other
+            try:
+                return Polynomial(*other._vector)
+            except AttributeError:
+                return Constant(other)
         elif not other:
-            return self
-        elif isinstance(other, Polynomial):
-            new_vector = []
-            max_iterations = max(self.degree, other.degree) + 1
-            for i in range(max_iterations):
-                a, b = 0, 0
-                try:
-                    a = self[i]
-                except IndexError:
-                    pass
-                try:
-                    b = other[i]
-                except IndexError:
-                    pass
-                new_vector.append(a+b)
-            return Polynomial(list(reversed(new_vector)))
-        else:
-            return self + Constant(other)
+            return Polynomial(*self._vector)
```

Comment:
`Polynomial(*self._vector)` does not create the same polynomial as it reverses the coefficients order. I would suggest using `Polynomial(self._vector[::-1])` instead.

Reply:
Same applies for `Polynomial(*other._vector)`